### PR TITLE
PYTHON-2512 Update Astrolabe's Workload Executor to use the unified test runner

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

**Comment by juliusgeo** [comment]
It's not quite ready for review as I still need to verify it is giving the correct results, however it looks like it works so far.

**Comment by juliusgeo** [comment]
@ShaneHarvey Should be ready for review.

File: integrations/python/pymongo/workload-executor.py
Comment (ShaneHarvey) [inline integrations/python/pymongo/workload-executor.py:47]: We should assert that test_workload["tests"] only has one test in it to make sure we're not silently ignoring extra tests.

File: integrations/python/pymongo/workload-executor.py
Comment (ShaneHarvey) [inline integrations/python/pymongo/workload-executor.py:45]: Remove this? Alternatively you could control this behavior via a command line argument to the workload-executor.py script.

File: integrations/python/pymongo/run-for-all.sh
Comment (ShaneHarvey) [inline integrations/python/pymongo/run-for-all.sh:6]: "user:password@localhost" is not a valid connection string because it's missing "mongodb://". How does this work?

File: integrations/python/pymongo/run-for-all.sh
Comment (ShaneHarvey) [inline integrations/python/pymongo/run-for-all.sh:1]: Make this the first line:
```
#!/bin/bash -ex
```

Also please add a comment that explains what this file is supposed to do and why it's useful.

File: integrations/python/pymongo/workload-executor.py
Comment (ShaneHarvey) [inline integrations/python/pymongo/workload-executor.py:25]: This method is unused.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: integrations/python/pymongo/workload-executor.py
Comment (juliusgeo) [reply integrations/python/pymongo/workload-executor.py]: Done.

File: integrations/python/pymongo/run-for-all.sh
Comment (juliusgeo) [reply integrations/python/pymongo/run-for-all.sh]: I am honestly not sure how it works. Could it mean that my routing in pymongo is incorrect?

File: integrations/python/pymongo/run-for-all.sh
Comment (juliusgeo) [reply integrations/python/pymongo/run-for-all.sh:4]: Done.

File: integrations/python/pymongo/workload-executor.py
Comment (juliusgeo) [reply integrations/python/pymongo/workload-executor.py:42]: Actually I don't think that that test proves my point. However, there is a section of the pseudocode implementation that I believe is what inspired my approach 
``` 
try {
        runner.executeScenario();
    } catch (propagatedError) {
        /* If the test runner propagates an error or failure (e.g. it is not
         * captured by the loop or occurs outside of the loop), it MUST be
         * reported by the workload executor. */
         errors.push({
           error: propagatedError.message,
           time: Date.now() / 1000
         });
    }
```

Comment (juliusgeo) [reply integrations/python/pymongo/workload-executor.py:42]: It also states this about error propagation:
"Note that is possible for some or all of these arrays to be empty if the corresponding data was not reported by the unified test runner and the test runner did not propagate an error or failure (which would then be reported by the workload executor)."

Comment (juliusgeo) [reply integrations/python/pymongo/workload-executor.py:42]: Also I tried it out without the error catching code and the spec says we MUST write out results.json and events.json, but there are assertion errors not being caught by pymongo that cause the program to exit without writing statistics.

Comment (juliusgeo) [reply integrations/python/pymongo/workload-executor.py:42]: An example of a test that causes the workload executor to throw an error (rather than it being caught in the loop) is /tests/validator-numFailures-not-captured.yml

File: .evergreen/config.yml
Comment (ShaneHarvey) [inline .evergreen/config.yml:356]: After merging PYTHON-3014, let's revert this.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: .evergreen/config.yml
Comment (juliusgeo) [reply .evergreen/config.yml]: Done.

File: integrations/python/pymongo/workload-executor.py
Comment (ShaneHarvey) [inline integrations/python/pymongo/workload-executor.py:12]: A comment that explains what and why this is required would be helpful.

**Review by ShaneHarvey — APPROVED** [review]
LGTM once the tests (mostly) pass. As discussed please open a new issue to investigate the flakey tests.

File: integrations/python/pymongo/workload-executor.py
Comment (juliusgeo) [inline integrations/python/pymongo/workload-executor.py:34]: This is necessary because there does not seem to exist a localhost:27017 instance of mongodb, so this instead overrides the client with the mongodb_uri that is where mongodb is actually located.

Comment (ShaneHarvey) [reply integrations/python/pymongo/workload-executor.py:34]: Please explain this in a comment as well.

Comment (juliusgeo) [reply integrations/python/pymongo/workload-executor.py:40]: Done.

File: integrations/python/pymongo/workload-executor.py
Comment (juliusgeo) [reply integrations/python/pymongo/workload-executor.py:15]: Done.

**Review by ShaneHarvey — APPROVED** [review]
LGTM

**Pull request merged**
